features/ui/sidebar-navigation/sidebar-navigation.tsx

```diff
@@ -153,22 +153,45 @@ const CollapseMenuItem = styled(MenuItemButton)<{ isCollapsed: boolean }>`
   }
 `;
 
+const useIsDesktop = (): boolean => {
+  const [isDesktop, setDesktop] = useState<boolean>(true);
+
+  const updateMedia = useCallback(() => {
+    setDesktop(window.innerWidth > 1023);
+  }, [setDesktop]);
+
+  useEffect(() => {
+    window.addEventListener("resize", updateMedia);
+    updateMedia();
+    return () => window.removeEventListener("resize", updateMedia);
+  });
+
+  return isDesktop;
+};
+
 export function SidebarNavigation() {
   const router = useRouter();
   const { isSidebarCollapsed, toggleSidebar } = useContext(NavigationContext);
   const [isMobileMenuOpen, setMobileMenuOpen] = useState(false);
+  const isDesktop = useIsDesktop();
+
+  const getImgSrc = (): string => {
```

Comment:
A bit of a nit-pick: I'd argue that it's better to make the `getImgSrc` function take two parameters e.g.

```
const getImgSrc = (isDesktop, isCollapsed) => {
  ....
}
```

This way it's a pure function that doesn't depend on any outside variables. Then you can also move it outside the component.

Reply:
I really appreciate all nit-picking! Great suggestion, I'll refactor it.